src/watertap_contrib/seto/solar_models/physical/flat_plate/irradiance_model_iso.py

```diff
@@ -0,0 +1,341 @@
+
+        self.G_T = Var(
+            initialize=1,
+            units=pyunits.W / pyunits.m**2,
+            bounds=(0, None),
+            doc="Total radiation on tilted surface",
+        )
+
+        self.B = Var(initialize=1, units=pyunits.degrees, doc="Solar B")
+
+        self.eqn_of_time = Var(
+            initialize=1, units=pyunits.minutes, doc="Equation of time"
+        )
+
+        self.G_trans = Var(
+            initialize=1,
+            units=pyunits.W / pyunits.m**2,
+            bounds=(0, None),
+            doc="Total radiation transmitted through glazing of tilted collector",
+        )
+
+        self.theta_ground = Var(
+            initialize=1,
+            units=pyunits.degrees,
+            bounds=(0, None),
+            doc="Effective incidence angle of ground reflected radiation [deg], D&B Eq. 5.4.1, pg. 213",
+        )
+
+        self.theta_diffuse = Var(
+            initialize=1,
+            units=pyunits.degrees,
+            bounds=(0, None),
+            doc="Effective incidence angle of sky diffuse radiation [deg], D&B Eq. 5.4.2, pg. 213",
+        )
+
+        self.Kta_b = Var(
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Incidence angle modifier coefficient for beam radiation",
+        )
+
+        self.Kta_d = Var(
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Incidence angle modifier coefficient for sky diffuse radiation",
+        )
+
+        self.Kta_g = Var(
+            initialize=1,
+            units=pyunits.dimensionless,
+            bounds=(0, None),
+            doc="Incidence angle modifier coefficient for ground reflected radiation",
+        )
+
+        # ==========CONSTRAINTS==========
+
+        @self.Constraint(doc="Equation for B, D&B eqn. 1.4.2")
+        def eq_B(b):
+            return self.B == (self.day_of_year - 1) * 360 * pyunits.deg / 365
+
+        @self.Constraint(doc="Solar time, D&B eqn. 1.5.2")
+        def eq_solar_time(b):
+            return self.solar_time == (
+                self.standard_time
+                + (
+                    4 * pyunits.minutes / pyunits.deg * (self.std_meridian - self.lon)
+                    + self.eqn_of_time
+                )
+                / (60 * pyunits.minutes / pyunits.hour)
+            )
+
+        @self.Constraint(doc="Equation of time, D&B eqn. 1.5.3")
+        def eq_eqn_of_time(b):
+            return self.eqn_of_time == (
+                229.2
+                * pyunits.minutes
+                * (
+                    0.000075
+                    + 0.001868 * cos(pyunits.convert(self.B, to_units=pyunits.rad))
+                    - 0.032077 * sin(pyunits.convert(self.B, to_units=pyunits.rad))
+                    - 0.014615 * cos(pyunits.convert(2 * self.B, to_units=pyunits.rad))
+                    - 0.04089 * sin(pyunits.convert(2 * self.B, to_units=pyunits.rad))
```

Comment:
We should convert vars/constraints to parameters/expressions to avoid usage of trigonometric functions with local solver.